File: emergency-centre-typescript/src/smoke-sensor.ts
import Action from "./action";
import Sensor from "./sensor";
import { Vendor } from "./vendor";

export default class SmokeSensor extends Sensor {
  private readonly sensitivity: number;

  constructor(
    id: number,
    vendor: Vendor,
    actions: Action[],
    sensitivity: number
  ) {
    super(id, vendor, actions);
    this.sensitivity = sensitivity;
  }

  get getSenstivity(): number {
    return this.sensitivity;
  }

  override toString(): string {
    return `
        SmokeSensor{id=${this.getId()}, vendor=${this.getVendor()}, 
        sensitivity=${this.sensitivity}}
        `;
  }
}

File: emergency-centre-c#/EmergencyCentre/SmokeSensor.cs
using System;
using EmergencyCentreDotNet.EmergencyCentre.Action;

namespace EmergencyCentreDotNet.EmergencyCentre;

public class SmokeSensor(int id, Vendor vendor, List<SensorAction> actions, double sensitivity) : Sensor(id, vendor, actions)
{
    public double Sensitivity { get; private set; } = sensitivity;

    public override string ToString()
    {
        return base.ToString() + $"_(sensitivity: {Sensitivity})";
    }
}
